lib/Core/BuildEngine.cpp

```diff
@@ -965,6 +965,8 @@ class BuildEngineImpl {
     
     // Find the cycle by searching from the entry node.
     struct WorkItem {
+      WorkItem(Rule * node) { this->node = node; }
+
```

Comment:
... interestingly, removing the default value of `predecessorIndex` allows MSVC to compile this just fine